libpromises/matching.c

```diff
@@ -292,7 +292,7 @@ void EscapeSpecialChars(const char *str, char *strEsc, int strEscSz, char *noEsc
                 break;
             }
 
-            strcat(strEsc, noEscSeq);
+            strlcat(strEsc, noEscSeq, strEscSz);
```

Comment:
But it is checked right above that it doesn't overflow, right? Or am I reading it wrong?